### DISCUSSION TITLE
Sunset/Sunrise mit variablem Offset an einem Pin ausgeben

### DISCUSSION THREAD

**Discussion by Andy-Op** [tbnobody/OpenDTU#1516] [Ideas]
Ich möchte folgenden Anwendungsfall von Sunrise/Sunset vorstellen: Dabei geht es mir darum tagsüber einen Akku zu laden und Nachts zur Grundlastkompensation wieder zu entladen. Ideal wäre es, die Entladung mit einem Offset zum Sonnenuntergang an einem Pin zu koppeln, Das Gleiche zum Zurücksetzen beim Sunrise. Alles was man dafür braucht ist im Programm/Prozessor vorhanden, Sunset, Sunrise und Geodaten sind bereits abgebildet. Es würde noch ein einstellbarer Offset und die Pin-Ausgabe fehlen. Leider fehlen mir die Möglichkeiten das programmtechnisch umzusetzen. Vielleicht findet @tbnobody hierzu Interesse.

**Comment by MichlW** [comment]
Ich kenne mich da jetzt nicht mega gut aus, aber gibt es überhaupt Batteriewechselrichter, die per digitalen oder analogen I/Os gesteuert werden? Dein Szenario lässt sich doch wesentlich flexibler in rein Software abbilden: Daten aus der DTU (sunrise/-set, aktuelle Leistung, ...) bestimmen, ob ein vorhandener Akku ge- oder entladen wird...
Wie gesagt, ich bin aus der Softwarewelt, aber dein UseCase klingt wie der Lösungsvorschlag eines Elektronikers.
Lohnt es sich dafür neue Schnittstellen auszudenken? Vermutlich findest du Unmengen an Howtos im Netz, wie du das alles über dein lokales Netzwerk ohne "neue Strippe ziehen" bereits realisieren kannst.
Disclaimer: Ich habe keinen Akku.
Viele Grüße, Michael

**Reply by Andy-Op** [reply]
Nicht so kompliziert denken. Ich sehe da 2 Relais mit 2 Sekunden Versatz schalten. Das bekomm ich auch analog hinter einem Optokoppler hin. Ich werde einen 120 Watt Inverter mir dem Labornetzteil testen. Alles was ich brauch ist ein Einschaltzeitpunkt. Also Sunset mit etwas Offset. Der Rest ist Energieerhaltungssatz und etwas Liebhaberei.

**Comment by Andy-Op** [comment]
Den Ablauf stelle ich mir wie folgt vor. Default, tagsüber low. Ab Sunset bis Sunrise high. Schaltung ist inaktiv bis high. Wechselrichter bekommt über sein Relais 230 Volt und DC vom Akku zugeschaltet. Miniinverter läuft an, Akku wird entladen. Ist Akku leer, entfällt die Betriebsspannung durch das BMS des Akkus. Relais fallen lastfrei ab. Kommt Sunrise früher wird zuerst netzseitig getrennt dann lastfrei DC abgeschaltet. Schaltung könnte wie folgt aussehen:

![Ausschaltverzögerung-DC-Relais](https://github.com/tbnobody/OpenDTU/assets/134189338/2b52bcdc-64d6-4d4a-a934-2877d14f0afd)

**Comment by Andy-Op** [comment]
Hier mal mein letzter Stand: Das unter Last schalten der Relais. insbesondere bei DC, sollte möglichst vermieden werden. Es wird eine Anlaufstrombegrenzung benötigt, die ich kontaktfrei mit einem NTC passend zu meinen Leistungswünschen realisiere. Der NTC pendelt sich beim Betriebsstrom bei etwa 0,1 Ohm ein und verbraucht bei 3,5 Ampere etwa so viel wie ein Relais. Der Wechseltrichter hängt primär mit seinem DC-Eingang am NTC. Das Stromnetz wird über ein SolideState-Relais nullpunktgeschaltet. Sanfter geht es nicht. Beim Ausschalten puffert R3&C2 das DC-Relais für einige Sekunden. Bis dahin hat der Wechselrichter abgeschaltet und das DC-Relais fällt lastfrei ab. Unterspannung fängt das BMS des Akkus ab.

![Relaissteuerung_Einspeisung_Balkonkraftwerk](https://github.com/tbnobody/OpenDTU/assets/134189338/4dc45a06-5b63-4e64-8e24-4836da2294c9)

Was fehlt ist eine astronomische Schaltuhr mit Offset. Sunrise und Sunset sind bereits im Code. Es fehlt ein einstellbarer Offset und die Pin-Ausgabe am ESP. Für hiesige Programmierer vielleicht 30 Minuten. Damit ermöglicht man eine ungeregelte low-cost-Einspeisung ohne zusätzliche Schaltuhren. Ich denke das müsste den Aufwand wert sein.

**Comment by Andy-Op** [comment]
AhoyDTU hat mit v0.8.42 (Weihnachts-Wunschliste) vorgelegt. Dort gibt es nun die astronomische Zeitschaltuhr mit einstellbarem Offset. Pins sogar frei definierbar. Den Optokoppler habe ich dran, sieht bisher gut aus. Ich mach dann erst mal da weiter.
